e2e-tests/verify-is-synced/src/main.rs

```diff
@@ -0,0 +1,262 @@
+    append_block(&block_4);
+
+    // Block 5: 10k transactions that transfer all of ADDRESS_2's BTC to ADDRESS_5
+    let mut block_5_txs = vec![];
+    for block_2_tx in block_2_txs {
+        block_5_txs.push(
+            TransactionBuilder::new()
+                .with_input(OutPoint {
+                    txid: block_2_tx.txid(),
+                    vout: 0,
+                })
+                .with_output(&Address::from_str(ADDRESS_5).unwrap(), 500_000)
+                .build(),
+        )
+    }
+
+    let mut block_5 = BlockBuilder::with_prev_header(block_4.header);
+    for tx in block_5_txs.into_iter() {
+        block_5 = block_5.with_transaction(tx);
+    }
+    let block_5 = block_5.build();
+    append_block(&block_5);
+
+    let next_block_1 = BlockBuilder::with_prev_header(block_5.header)
+        .with_transaction(
+            TransactionBuilder::new()
+                .with_output(&Address::from_str(ADDRESS_5).unwrap(), 500_000)
+                .build(),
+        )
+        .build();
+    append_block_header(&next_block_1.header);
+    let next_block_2 = BlockBuilder::with_prev_header(next_block_1.header)
+        .with_transaction(
+            TransactionBuilder::new()
+                .with_output(&Address::from_str(ADDRESS_5).unwrap(), 500_000)
+                .build(),
+        )
+        .build();
+    append_block_header(&next_block_2.header);
+    let next_block_3 = BlockBuilder::with_prev_header(next_block_2.header)
+        .with_transaction(
+            TransactionBuilder::new()
+                .with_output(&Address::from_str(ADDRESS_5).unwrap(), 500_000)
+                .build(),
+        )
+        .build();
+    append_block_header(&next_block_3.header);
+}
+
+#[update]
+fn bitcoin_get_successors(request: GetSuccessorsRequest) -> GetSuccessorsResponse {
+    if let GetSuccessorsRequest::Initial(GetSuccessorsRequestInitial { network, .. }) = &request {
+        assert_eq!(
+            *network,
+            Network::Regtest,
+            "request must be set to the regtest network"
+        );
+    }
+
+    let count = COUNT.with(|c| c.get());
+
+    let res = if count == 0 {
+        // Send block 1 in full.
+        GetSuccessorsResponse::Complete(GetSuccessorsCompleteResponse {
+            blocks: vec![BLOCKS.with(|b| b.borrow()[0].clone())],
+            next: vec![],
+        })
+    } else if count == 1 {
+        // Send part of block 2.
+        GetSuccessorsResponse::Partial(GetSuccessorsPartialResponse {
+            partial_block: BLOCKS.with(|b| b.borrow()[1].clone())[0..20].to_vec(),
+            next: vec![],
+            remaining_follow_ups: 2,
+        })
+    } else if count == 2 {
+        // Send another part of block 2.
+        GetSuccessorsResponse::FollowUp(BLOCKS.with(|b| b.borrow()[1].clone())[20..40].to_vec())
+    } else if count == 3 {
+        // Send rest of block 2.
+        GetSuccessorsResponse::FollowUp(BLOCKS.with(|b| b.borrow()[1].clone())[40..].to_vec())
```

Comment:
Splitting the block across multiple responses isn't being tested in this scenario, so I'd just return the complete response.